src/overlays/actors/ovl_En_Xc/z_en_xc.c

```diff
@@ -77,479 +47,2365 @@ static ColliderCylinderInitType1 sCylinderInit = {
+        startFrame = frameCount;
+        endFrame = 0.0f;
+        playbackSpeed = -1.0f;
+    }
+
+    Animation_Change(&this->skelAnime, animationSeg, playbackSpeed, startFrame, endFrame, mode, morphFrames);
+}
+
+void EnXc_CheckAndSetAction(EnXc* this, s32 check, s32 set) {
+    if (check != this->action) {
+        this->action = set;
+    }
+}
+
+void func_80B3C7D4(EnXc* this, s32 action1, s32 action2, s32 action3) {
+    if (action1 != this->action) {
+        if (this->action == SHIEK_ACTION_PUT_HARP_AWAY) {
+            this->action = action2;
+        } else {
+            this->action = action3;
+        }
+    }
+}
+
+s32 EnXc_NoCutscenePlaying(GlobalContext* globalCtx) {
+    if (globalCtx->csCtx.state == 0) {
+        return true;
+    }
+    return false;
+}
+
+void func_80B3C820(EnXc* this) {
+    Animation_Change(&this->skelAnime, &gShiekIdleAnim, 1.0f, 0.0f, Animation_GetLastFrame(&gShiekIdleAnim),
+                     ANIMMODE_LOOP, 0.0f);
+    this->action = SHIEK_ACTION_53;
+}
+
+void func_80B3C888(EnXc* this, GlobalContext* globalCtx) {
+    if (EnXc_NoCutscenePlaying(globalCtx) && this->actor.params == SHIEK_TYPE_4) {
+        func_80B3C820(this);
+    }
+}
+
+void func_80B3C8CC(EnXc* this, GlobalContext* globalCtx) {
+    SkelAnime* skelAnime = &this->skelAnime;
+
+    if (skelAnime->jointTable[0].y >= skelAnime->baseTransl.y) {
+        skelAnime->moveFlags |= 3;
+        AnimationContext_SetMoveActor(globalCtx, &this->actor, skelAnime, 1.0f);
+    }
+}
+
+void func_80B3C924(EnXc* this, GlobalContext* globalCtx) {
+    this->skelAnime.moveFlags |= 3;
+    AnimationContext_SetMoveActor(globalCtx, &this->actor, &this->skelAnime, 1.0f);
+}
+
+void func_80B3C964(EnXc* this, GlobalContext* globalCtx) {
+    this->skelAnime.baseTransl = *this->skelAnime.jointTable;
+    this->skelAnime.prevTransl = *this->skelAnime.jointTable;
+    this->skelAnime.moveFlags |= 3;
+    AnimationContext_SetMoveActor(globalCtx, &this->actor, &this->skelAnime, 1.0f);
+}
+
+void func_80B3C9DC(EnXc* this) {
+    this->skelAnime.moveFlags &= ~0x3;
+}
+
+void func_80B3C9EC(EnXc* this) {
+    EnXc_ChangeAnimation(this, &gShiekArmsCrossedIdleAnim, ANIMMODE_LOOP, 0.0f, false);
+    this->action = SHIEK_ACTION_BLOCK_PEDESTAL;
+    this->drawMode = SHIEK_DRAW_DEFAULT;
+    this->unk_30C = 1;
+}
+
+void func_80B3CA38(EnXc* this, GlobalContext* globalCtx) {
+    // If Player is adult but hasn't learned Minuet of Forest
+    if (!(gSaveContext.eventChkInf[5] & 1) && LINK_IS_ADULT) {
+        this->action = SHIEK_ACTION_INIT;
+    } else {
+        Actor_Kill(&this->actor);
+    }
+}
+
+s32 EnXc_MinuetCS(EnXc* this, GlobalContext* globalCtx) {
+    if (this->actor.params == SHIEK_TYPE_MINUET) {
+        Player* player = PLAYER;
+        f32 z = player->actor.world.pos.z;
+        if (z < -2225.0f) {
+            if (!Gameplay_InCsMode(globalCtx)) {
+                globalCtx->csCtx.segment = SEGMENTED_TO_VIRTUAL(&spot05_sceneCutsceneData0x003F80);
```

Comment:
This cutscene reference should be named properly instead of using the auto-generated ZAP name.
Relevant for all the other cutscene references in this file